providers/openmetrics/go.mod

```diff
@@ -11,3 +11,5 @@ require (
 	google.golang.org/grpc v1.35.0
 	google.golang.org/protobuf v1.25.0
 )
+
+replace github.com/grpc-ecosystem/go-grpc-middleware/v2 => ../..
```

Comment:
I'm not sure how to test it otherwise if changes are made to the outer module and to providers.

Reply:
I think it'll have to be merged in two steps.

Reply:
If this PR is approved I can split it into two - one updating the base and another one updating the provider. I also have a few more things I'd like to cleanup in follow ups.

Reply:
Sounds good to me, will give @bwplotka a few days to have a look but I'm otherwise happy to merge this on our own.

Reply:
Yup, this is the problem with multi-modules.

One thing we could add is to add some tooling that e.g makes commit with this PR on your fork, then creates commit on top of it, so it's at least not broken on master, if you just pull openmetrics module. Local path approach can break users on main. It's fine for now (dev part of v2) but we might want to improve this process. LGTM for now